File: JAVA/731. My Calendar II.java
class MyCalendarTwo {
    List<int[]> bookings;
    List<int[]>overlappingIntervals;
    public MyCalendarTwo() {
        bookings=new ArrayList<>();
        overlappingIntervals=new ArrayList<>();
    }    
    public boolean book(int start, int end) {
        for(int[] i:overlappingIntervals)
        {
            if(start<i[1]  && end>i[0])
            {
                return false;
            }
        }       
        for(int[] b:bookings)
        {
            if(start<b[1] && end>b[0])
            {
                overlappingIntervals.add(new int[]{Math.max(start,b[0]),Math.min(end,b[1])});
            }
        }
        bookings.add(new int[]{start,end});
        return true;
    }
}

File: C++/731. My Calendar II.cpp
class MyCalendarTwo {
private:
    vector<pair<int, int>> bookings;
    vector<pair<int, int>> overlaps;
public:
    MyCalendarTwo() {        
    }    
    bool book(int start, int end) {
        for (const auto& [s, e] : overlaps) {
            if (start < e && end > s) {
                return false;
            }
        }
        for (const auto& [s, e] : bookings) {
            if (start < e && end > s) {
                overlaps.push_back({max(start, s), min(end, e)});
            }
        }
        bookings.push_back({start, end});
        return true;
    }
};
